app/controllers/frontend_log_controller.rb

```diff
@@ -77,4 +77,15 @@ def valid_event?
   def valid_payload?
     !log_params[:payload].nil?
   end
+
+  def hash_from_method_kwargs(hash, method)
+    method_kwargs(method).index_with { |key| hash[key.to_s] }
+  end
+
+  def method_kwargs(method)
+    method.
+      parameters.
+      select { |info_pair| [:key, :keyreq].include?(info_pair[0]) }.
+      map(&:last)
```

Comment:
I have two ideas for this block

The first gets rid of `[0]` by destructuring

```suggestion
    method.
      parameters.
      select { |type, _name| [:key, :keyreq].include?(type) }.
      map(&:last)
```

the second goes for a map + compact route

```suggestion
    method.
      parameters.
      map do |type, name|
        name if [:key, :keyreq].include?(type)
      end.compact
```

Reply:
I don't think reduce is such a great fit here, but anything we do to name the params, I'm in favor of. I think these arrays are small enough that it's OK to do 2 passes

Reply:
Fair enough, I'll go with `map` + `compact`.